src/main/java/org/jenkinsci/plugins/GithubSecurityRealm.java

```diff
@@ -400,6 +411,23 @@ public HttpResponse doFinishLogin(StaplerRequest request)
         return HttpResponses.redirectToContextRoot();   // referer should be always there, but be defensive
     }
 
+    private void storeCredentials(String accessToken, GithubAuthenticationToken auth) throws IOException {
+        CredentialsStore credentialsStore = CredentialsProvider.lookupStores(Jenkins.getInstance()).iterator().next();
```

Comment:
Oh my god NOOOOO!

`CredentialsProvider.lookupStore(user)` (or something like that, I'm on a phone)

You are storing the user's credentials in the global store otherwise... plus there may be more than one global store, and the first one may not be writable